trunk/SUAVE/Methods/Flight_Dynamics/Static_Stability/compute_aero_derivatives.py

```diff
@@ -50,47 +49,71 @@ def compute_aero_derivatives(segment, h=1e-4):
     # Alpha perturbation
     
     perturbed_segment = deepcopy(segment)    
-    alpha_plus        = alpha*(1+h)
-    perturbed_segment.state.conditions.frames.body.inertial_rotations[:,1] = alpha_plus
+    pitch_plus        = pitch*(1+h)
+    perturbed_segment.state.conditions.frames.body.inertial_rotations[:,1] = pitch_plus
     
     iterate = perturbed_segment.process.iterate
     iterate.conditions(perturbed_segment) 
     
     # set segment derivatives based on perturbed segment
-    dAlpha = np.atleast_2d(alpha_plus-alpha).T
+    dAlpha = perturbed_segment.state.conditions.aerodynamics.angle_of_attack - segment.state.conditions.aerodynamics.angle_of_attack
     dCL    = perturbed_segment.state.conditions.aerodynamics.lift_coefficient - segment.state.conditions.aerodynamics.lift_coefficient
     dCD    = perturbed_segment.state.conditions.aerodynamics.drag_coefficient - segment.state.conditions.aerodynamics.drag_coefficient
+    dCM    = perturbed_segment.state.conditions.stability.static.CM - segment.state.conditions.stability.static.CM
 
+    # propeller derivatives
+    dCT, dCP = propeller_derivatives(segment, perturbed_segment, n_cpts)
+        
     dCL_dAlpha = dCL/dAlpha
     dCD_dAlpha = dCD/dAlpha
+    dCM_dAlpha = dCM/dAlpha
+    dCT_dAlpha = dCT/dAlpha[None,:,:]
+    dCP_dAlpha = dCP/dAlpha[None,:,:]
     
     segment.state.conditions.aero_derivatives.dCL_dAlpha = dCL_dAlpha
     segment.state.conditions.aero_derivatives.dCD_dAlpha = dCD_dAlpha
+    segment.state.conditions.aero_derivatives.dCM_dAlpha = dCM_dAlpha
+    segment.state.conditions.aero_derivatives.dCT_dAlpha = dCT_dAlpha
+    segment.state.conditions.aero_derivatives.dCP_dAlpha = dCP_dAlpha
     
     
     # ----------------------------------------------------------------------------    
     # Beta perturbation
     
     perturbed_segment = deepcopy(segment)    
-    beta_plus         = beta+h
-    perturbed_segment.state.conditions.frames.body.inertial_rotations[:,2] = beta_plus
+    psi_plus          = psi+h
+    perturbed_segment.state.conditions.frames.body.inertial_rotations[:,2] = psi_plus
     
     iterate = perturbed_segment.process.iterate
     iterate.conditions(perturbed_segment) 
     
     # set segment derivatives based on perturbed segment
-    dBeta  = np.atleast_2d(beta_plus-beta).T
+    dBeta  = perturbed_segment.state.conditions.aerodynamics.side_slip_angle - segment.state.conditions.aerodynamics.side_slip_angle #np.atleast_2d(psi_plus-psi).T
     dCL    = perturbed_segment.state.conditions.aerodynamics.lift_coefficient - segment.state.conditions.aerodynamics.lift_coefficient
     dCD    = perturbed_segment.state.conditions.aerodynamics.drag_coefficient - segment.state.conditions.aerodynamics.drag_coefficient
+    dCM    = perturbed_segment.state.conditions.stability.static.CM - segment.state.conditions.stability.static.CM
```

Comment:
This is a meaningless derivative

Reply:
[See here for the derivates we get out of the VLM](https://github.com/suavecode/SUAVE/blob/e8183535680adc9fec07ffc57499640ff12f79af/trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/VLM.py#L494)

They're in VORLAX notation and not in standard notation. Let me know if you need help decoding them.